react-hot-button-component/src/hot-button.jsx

```diff
@@ -0,0 +1,31 @@
+import React from 'react';
+
+let counter = 0;
+let statusTracker = 0;
+const colors = ['purple', 'light-purple', 'red', 'orange', 'yellow', 'white'];
+
+class HotButton extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      status: null // default color
```

Comment:
The thing that changes over time is the number of times the button has been clicked, not the color.